libs/execution_context/ebpf_native.c

```diff
@@ -1451,11 +1456,11 @@ _ebpf_native_clean_up_handle_cleanup_context(_Inout_ ebpf_native_handle_cleanup_
     }
 
     if (cleanup_context->handle_cleanup_work_item != NULL) {
-        // ebpf_free_preemptible_work_item() will free the handle_information.
-        ebpf_free_preemptible_work_item(cleanup_context->handle_cleanup_work_item);
-    } else {
-        ebpf_free(cleanup_context->handle_information);
+        cxplat_free_preemptible_work_item(cleanup_context->handle_cleanup_work_item);
+        cleanup_context->handle_cleanup_work_item = NULL;
```

Comment:
Could we merge nulling the pointer into `cxplat_free_preemptible_work_item`?

Reply:
it'd be different from the normal POSIX convention.  @nibanks any opinion?

Reply:
Can we make this an issue and get this PR merged?

Reply:
@Alan-Jowett @dthaler I'm fine to eventually to track this into another issue.

Reply:
It's not clear whether it's a good idea, but filed https://github.com/microsoft/usersim/issues/112 to track that discussion.